controllers/database_controller.py

```diff
@@ -2,71 +2,45 @@
 
 import requests
 from sqlalchemy import create_engine
+from sqlalchemy.exc import IntegrityError
 from structlog import wrap_logger
 
 from config import Config
 
 logger = wrap_logger(logging.getLogger(__name__))
 
 
-def execute_rm_sql(sql_script_file_path):
-    logger.info('Executing SQL script', sql_script=sql_script_file_path)
-
-    url = Config.CF_DATABASE_TOOL + '/sql'
-    headers = {
-        'Content-Type': 'text/plain'
-    }
-    with open(sql_script_file_path, 'r') as sqlScriptFile:
-        sql_script = sqlScriptFile.read().replace('\n', '')
-
-    response = requests.post(url, auth=Config.BASIC_AUTH, headers=headers, data=sql_script)
-
-    # The response from the database tool as of 16/02/18 always comes back with 201 regardless
-    # of whether the sql execution succeded or failed.  If tests are failing, check the database tool's
-    # logs for any sqlExceptions.
-    if response.status_code != 201:
-        logger.error('SQL execution failed', status=response.status_code, sql_script=sql_script_file_path)
-
-    logger.info('Executed SQL script', sql_script=sql_script_file_path)
-    return response.text
-
-
-def reset_ras_database():
-    logger.info("Putting party database into known state")
-    engine = create_engine(Config.PARTY_DATABASE_URI)
+def execute_ras_sql(sql_script_file_path, database_uri=Config.DATABASE_URI):
+    logger.debug('Executing SQL script', sql_script_file_path=sql_script_file_path)
+    engine = create_engine(database_uri)
     connection = engine.connect()
     trans = connection.begin()
 
-    with open('resources/database/database_reset_party.sql', 'r') as sqlScriptFile:
-        reset_party_sql = sqlScriptFile.read().replace('\n', '')
-
-    connection.execute(reset_party_sql)
-    trans.commit()
-
-    logger.info("Putting django database into known state")
-    engine = create_engine(Config.DJANGO_OAUTH_DATABASE_URI)
-    connection = engine.connect()
-    trans = connection.begin()
+    with open(sql_script_file_path, 'r') as sqlScriptFile:
+        sql = sqlScriptFile.read().replace('\n', '')
 
-    with open('resources/database/database_reset_oauth.sql', 'r') as sqlScriptFile:
-        reset_oauth_sql = sqlScriptFile.read().replace('\n', '')
+    try:
+        connection.execute(sql)
+    except IntegrityError:
+        logger.info('Script has already been run', sql_script_file_path=sql_script_file_path)
 
-    connection.execute(reset_oauth_sql)
     trans.commit()
+    logger.debug('Successfully executed SQL script', sql_script_file_path=sql_script_file_path)
 
 
-def reset_secure_message_database():
-    logger.info("Clearing down secure message database")
-    engine = create_engine(Config.SECURE_MESSAGE_DATABASE_URI)
-    connection = engine.connect()
-    trans = connection.begin()
-
-    with open('resources/database/database_reset_secure_message.sql', 'r') as sqlScriptFile:
+def execute_rm_sql(sql_script_file_path):
```

Comment:
Is it worth running this like you've done above?

Reply:
So apparently in the cf environments the different databases have different username/passwords required in the URI. That's why we pass in a specific database URI into the execute_ras_sql, they'll be different for all the RAS databases. For the RM databases we get around this problem by using the cf database tool.

I'd like it to be more consistent ideally but I don't think it's easy to do with the current database setup in cloudfoundry

Reply:
I do have a couple more changes I'll push to this file to make it a bit more generic though before the PR goes through